go/ql/lib/semmle/go/security/IncorrectIntegerConversionLib.qll

```diff
@@ -288,23 +411,50 @@ class UpperBoundCheckGuard extends DataFlow::RelationalComparisonNode {
       then strictnessOffset = 1
       else strictnessOffset = 0
     |
-      (
-        bound = expr.getAnOperand().getExactValue().toFloat()
-        or
-        exists(DeclaredConstant maxint | maxint.hasQualifiedName("math", "MaxInt") |
-          expr.getAnOperand() = maxint.getAReference() and
-          bound = getMaxIntValue(32, true)
-        )
-        or
-        exists(DeclaredConstant maxuint | maxuint.hasQualifiedName("math", "MaxUint") |
-          expr.getAnOperand() = maxuint.getAReference() and
-          bound = getMaxIntValue(32, false)
-        )
+      exists(DeclaredConstant maxint, DeclaredConstant maxuint |
+        maxint.hasQualifiedName("math", "MaxInt") and maxuint.hasQualifiedName("math", "MaxUint")
+      |
+        if expr.getAnOperand() = maxint.getAReference()
+        then bound = getMaxIntValue(32, true)
+        else
+          if expr.getAnOperand() = maxuint.getAReference()
+          then bound = getMaxIntValue(32, false)
+          else bound = expr.getAnOperand().getExactValue().toFloat()
       ) and
       bound - strictnessOffset <= getMaxIntValue(bitSize, isSigned)
     )
   }
 
+  /**
+   * Holds if this upper bound check ensures the non-constant operand is less
+   * than or equal to `2^(bitsize) - 1`. In this  case, the upper bound check
+   * is a barrier guard.
```

Comment:
Mention what `architectureBitSize` is doing here, and the distinction from `isBoundFor`